src/account/services/update-project.service.ts

```diff
@@ -1,117 +1,85 @@
         });
 
         dbProject.accessAddresses =
-            updatedProject.accessAddresses?.map((address) =>
-                address.toLowerCase(),
-            ) ?? dbProject.accessAddresses;
-
-        dbProject.description =
-            updatedProject.description ?? dbProject.description;
+            updatedProject.accessAddresses?.map((address) => address.toLowerCase()) ??
+            dbProject.accessAddresses;
 
+        dbProject.description = updatedProject.description ?? dbProject.description;
         dbProject.oneLiner = updatedProject.oneLiner ?? dbProject.oneLiner;
-
         dbProject.category = updatedProject.category ?? dbProject.category;
+        dbProject.socialMedia = updatedProject.socialMedia ?? dbProject.socialMedia;
 
-        dbProject.socialMedia =
-            updatedProject.socialMedia ?? dbProject.socialMedia;
+        if (updatedProject.deletedImages) {
+            for (const imageId of updatedProject.deletedImages) {
+                for (const [index, dbImage] of dbProject.images.entries()) {
+                    if (dbImage.id === imageId) {
+                        await this.s3ImageManagementService.deleteFileOnS3(dbImage as Image);
 
-        if (updatedProject.deletedPictures) {
-            for (const pictureKey of updatedProject.deletedPictures) {
-                for (const [index, dbPicture] of dbProject.pictures.entries()) {
-                    if (dbPicture.key === pictureKey) {
-                        await this.s3ImageManagementService.deleteFileOnS3(
-                            dbPicture,
-                        );
-
-                        dbProject.pictures.splice(index, 1);
+                        dbProject.images.splice(index, 1);
                     }
                 }
             }
         }
 
-        if (updatedProject.newPictures) {
-            if (!dbProject.pictures) {
-                dbProject.pictures = [];
+        if (updatedProject.newImages) {
+            if (!dbProject.images) {
+                dbProject.images = [];
             }
 
-            for (const picture of updatedProject.newPictures) {
-                const optimizedPicture =
-                    await this.picturesService.optimizeGalleryImage({
-                        data: picture.buffer,
-                        type: picture.mimetype as MimeTypesEnum,
-                    });
-
-                dbProject.pictures.push(
-                    await this.s3ImageManagementService.uploadImageToS3(
-                        optimizedPicture.data,
-                        optimizedPicture.type,
-                        dbProject.id,
-                    ),
-                );
+            for (const image of updatedProject.newImages) {
+                const unassignedImage = await this.imageRepository.findOneRaw({
+                    find: { id: image },
+                });
+
+                dbProject.images = dbProject.images as Image[];
+                dbProject.images.push(unassignedImage);
```

Comment:
question: Do we have an attack vector here? If a bad project links image id's from another project. The bad project could call a delete for those images. The images would disappear for the other project as well. Shouldn't we check if an image has already been linked?

Reply:
good point, i will add a check for that!